sklearn/utils/estimator_checks.py

```diff
@@ -397,6 +398,39 @@ def check_dtype_object(name, Estimator):
 
 
 @ignore_warnings
+def check_dict(name, Estimator):
+    rnd = np.random.RandomState(0)
+    if name in ['SpectralCoclustering']:
+        return 0
+    if name in ['RANSACRegressor']:
+        X = 3 * rnd.uniform(size=(20, 3))
+    else:
+        X = 2 * rnd.uniform(size=(20, 3))
+
+    y = X[:, 0].astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    estimator = Estimator()
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 3
```

Comment:
well yes, but for some reason with `1` I catch `ValueError: n_best cannot be larger than n_components, but 3 >  1` from `SpectralBiclustering`